modules/git/command_test.go

```diff
@@ -26,4 +26,21 @@ func TestRunWithContextStd(t *testing.T) {
 		assert.Contains(t, err.Error(), "exit status 129 - unknown option:")
 		assert.Empty(t, stdout)
 	}
+
+	assert.Panics(t, func() {
+		cmd = NewCommand(context.Background())
+		cmd.AddDynamicArguments("-test")
+	})
+
+	assert.Panics(t, func() {
+		cmd = NewCommand(context.Background())
+		cmd.AddDynamicArguments("--test")
+	})
+
+	subCmd := "version"
+	cmd = NewCommand(context.Background()).AddDynamicArguments(subCmd) // for test purpose only, the sub-command should never be dynamic for production
```

Comment:
The git command module should be refactored later.

Reply:
I meant something like `Do not call this when you don't know which git command to call` as comment for the new function.

Reply:
Maybe people do not like this PR. If there are approvals it can be improved.

Reply:
I have absolutely nothing against approving it, I simply didn't quite understand why we panic instead of returning an error as normally done in such situations.